packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -71,12 +71,22 @@ interface AutoSize extends BaseProps {
 
 interface Collapsible {
   collapsible: boolean;
+  /**
+   * used to set collapsed or expanded state on initial render
+   */
+  defaultExpanded?: boolean;
```

Comment:
Why don't we just pass if it is collapsed?
```
collapsed={true|false}
```
So we can make the panel a controlled state.

```jsx
function MyPage(){
  const [collapsed, setCollapsed]
  return (
    <PanelChrome
      collapsible
      collapsed={collapsed}
      onToggleCollapse={(collapsed)=>setCollapsed(collapsed)}
  )
}
```
I have to say, it is weird to have `collapsible` and `collapsed` as part of the same component API.
What do we do if we pass `collapsed` and `onToggleCollapse` but not `collapsible`?

How it should work:
1. If we pass `collapsed` it becomes collapsible but it doesn't change the state when clicking because the component becomes controlled. Like an input when you pass a `value`
2. If we pass `onToggleCollapse` but not collapsed, it doesn't become collapsible
3. If we pass `collapsed` and `onToggleCollapse` the component becomes collapsible, so the prop `collapsible` is not required
4. If we pass only `collapsible`, the component is uncontrolled, so it collapse by itself, with the internal state. If we pass the callback, the callback is called, but the state is internal until we don't pass `collapsible` as prop.

Is it clear? Sorry, quite hard to put everything on the table 😄

Reply:
@ivanortegaalba thank you for your feedback on this. 

I understand what you are saying and here some clarifying comments and questions:

> What do we do if we pass `collapsed` and `onToggleCollapse` but not `collapsible`?

This wouldn't be possible because of how the `Collapsible` interface is defined. `collapsible` always has to be passed if `defaultExpanded` or `onToggleCollapse` are to be used.

> 1. If we pass collapsed it becomes collapsible but it doesn't change the state when clicking because the component becomes controlled. Like an input when you pass a value
> 2. If we pass `onToggleCollapse` but not collapsed, it doesn't become collapsible

Why would we pass `onToggleCollapse` or 'collapsed` just by themselves if the component is not collapsible? 

In my mind, I think it's more readable to first declare if component is collapsible, and then pass other props related to that property, like if the component should be collapsed on first render and then pass any callbacks on toggling.

Reply:
Thanks for the answer @harisrozajac 

> This wouldn't be possible because of how the Collapsible interface is defined. collapsible always has to be passed if defaultExpanded or onToggleCollapse are to be used.

Yep, I understand this would be a breaking change, because `collapsible` is already released.

> Why would we pass onToggleCollapse or collapsed just by themselves if the component is not collapsible?

When you pass them, you're implicitly declaring the component as collapsible. In the same way when you declare `onClick` you are making the component clickable but you don't add a property called `clickable`.

> In my mind, I think it's more readable to first declare if component is collapsible, and then pass other props related to that property, like if the component should be collapsed on first render and then pass any callbacks on toggling.

My suggestion is not about readability, is more about standardization. When you have an input text or any other component that can receive a state from outside, you don't rely on an internal state, this is what is known as [a controlled component](https://legacy.reactjs.org/docs/forms.html#controlled-components).

With this API, you cannot make the component controlled, since the component has an internal state that you cannot force or change from outsize. You would need a full reset of the component to change the state from outside. For example, if I want to have a keyboard shortcut that toggles all the panels, how would you implement it? You must apply the logic inside the component. Or if you want to open the panel when another element is clicked, you cannot make it happen. 

This is why I miss the possibility to make the component controlled, and you can only do that by allowing `collapsed` property. But if you allow `collapsed`, `defaultExpanded` doesn't make sense anymore.

So, my new suggestion, to combine your approach with the controlled component is:
```jsx
function MyPage(){
  // By default will be open
  const [isCollapsed, setCollapsed] = useState(true)

  return (
    <PanelChrome
      {/* This must be provided to make the component collapsible */}
      collapsible
      {/* This will expand or collapse no matter the internal state. If it is not provided, the component will be expanded by default, and it will have the toggle behavior as usual */}
      collapsed={isCollapsed}
      {/* This will receive the new state, toggling the previous one */}
      onToggleCollapse={(isCollapsed)=>setCollapsed(isCollapsed)}
  ) />
}
```

What do you think?

Reply:
@ivanortegaalba thank you for the clarification. It makes sense and I'll make the adjustments 😄